### PR TITLE
Support ActiveSupport 5

### PR DESCRIPTION
**Pull request by Empact**
This builds on #48 to get a green build under activesupport 4 and 5.

### CODE REVIEW

File: test/active_support/cache/redis_store_test.rb [hunk 1]
```diff
@@ -508,12 +514,25 @@ def with_store_management
...
-        @events << ActiveSupport::Notifications::Event.new(*args)
+      instrument_store do
+        ActiveSupport::Notifications.subscribe(/^cache_(.*)\.active_support$/) do |*args|
+          @events << ActiveSupport::Notifications::Event.new(*args)
+        end
+        yield
+      end
+    end
+
+    if ActiveSupport::VERSION::MAJOR < 5
```
Comment (tubbo) [inline test/active_support/cache/redis_store_test.rb]: why not make `def instrument_store` do the conditional inside the method so you're not conditionally defining a method?

File: test/active_support/cache/redis_store_test.rb [hunk 2]
```diff
@@ -508,12 +514,25 @@ def with_store_management
 
     def with_notifications
       @events = [ ]
-      ActiveSupport::Cache::RedisStore.instrument = true
-      ActiveSupport::Notifications.subscribe(/^cache_(.*)\.active_support$/) do |*args|
-        @events << ActiveSupport::Notifications::Event.new(*args)
+      instrument_store do
```
Comment (tubbo) [inline test/active_support/cache/redis_store_test.rb]: actually this whole method seems wrong...can't it just be a nested conditional?

**Comment by tubbo** [comment]
hey @Empact, had a couple comments for the redis_store_test but other than that, looks good. will merge along with https://github.com/redis-store/redis-actionpack/pull/8 to support rails 5 in redis-rails.

**Comment by tubbo** [comment]
hey @Empact what's the deal with this? You still gonna work on it or should I re-assign?

**Comment by Empact** [comment]
Please reassign - I should have time to work on it in the next week or so
most likely but would rather not be the long pole.

**Comment by troelskn** [comment]
Hi. Should I finish up the work on this one, so it can be merged? E.g. refactor the conditional code, so that it is moved into `instrument_store` instead of `with_notifications`, unless I'm missing something?

**Comment by dreyks** [comment]
#57

**Pull request closed without merging**